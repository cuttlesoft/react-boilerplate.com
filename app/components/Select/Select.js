/* eslint-disable react/prop-types */
import React from 'react'
import { Select } from 'grommet'

import { doc } from './Select.doc'
import helpers from '../../utils/helpers'

/**
 *
 * Select
 *
 */
function createSelect({
  a11yTitle,
  alignSelf,
  children,
  closeOnChange,
  disabled,
  dropAlign,
  dropProps,
  dropHeight,
  dropTarget,
  emptySearchMessage,
  focusIndicator,
  labelKey,
  icon,
  margin,
  messages,
  multiple,
  onChange,
  onClose,
  onMore,
  onOpen,
  onSearch,
  open,
  options,
  placeholder,
  plain,
  replace,
  searchPlaceholder,
  selected,
  size,
  value,
  ...rest
}) {
  return (
    <Select
      a11yTitle={a11yTitle}
      alignSelf={alignSelf}
      closeOnChange={closeOnChange}
      disabled={disabled}
      dropAlign={dropAlign}
      dropProps={dropProps}
      dropHeight={dropHeight}
      dropTarget={dropTarget}
      emptySearchMessage={emptySearchMessage}
      focusIndicator={focusIndicator}
      labelKey={labelKey}
      icon={icon}
      margin={margin}
      messages={messages}
      multiple={multiple}
      onChange={onChange}
      onClose={onClose}
      onOpen={onOpen}
      onSearch={onSearch}
      open={open}
      options={options}
      placeholder={placeholder}
      plain={plain}
      replace={replace}
      searchPlaceholder={searchPlaceholder}
      selected={selected}
      size={size}
      value={value}
      {...rest}
    >
      {children}
    </Select>
  )
}

export default helpers.createWithDoc({
  envName: process.env.NODE_ENV,
  docFunction: doc,
  component: createSelect,
})
