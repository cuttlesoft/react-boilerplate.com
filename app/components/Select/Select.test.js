import React from 'react'
import { render, fireEvent } from '@testing-library/react'

import Select from './Select'

const options = ['one', 'two', 'three']
const renderComponent = (props = {}) =>
  render(<Select label="Select" {...props} options={options} />)

const setup = () => {
  const utils = render(<Select placeholder="select" options={options} />)
  const select = utils.getByPlaceholderText('select')
  return {
    select,
    ...utils,
  }
}

/**
 *
 * Tests for Select
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */
describe('Select', () => {
  it('renders and matches snapshot', () => {
    const { container } = renderComponent()

    expect(container).toMatchSnapshot()
  })

  it('should update value on select', () => {
    const { select } = setup()

    expect(select.value).toBe('')

    fireEvent.select(select, { target: { value: 'one' } })

    expect(select.value).toBe('one')
  })
})
