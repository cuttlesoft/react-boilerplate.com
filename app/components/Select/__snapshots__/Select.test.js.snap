// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders and matches snapshot 1`] = `
<div>
  <button
    aria-label="Open Drop"
    class="StyledButton-sc-323bzc-0 ebqXLG Select__StyledSelectDropButton-sc-17idtfo-1 iGEQF"
    type="button"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 jmuHez"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 iAryDB"
      >
        <div
          class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 dpaJQp"
        >
          <input
            autocomplete="off"
            class="StyledTextInput-sc-1x30a0s-0 fXMpul Select__SelectTextInput-sc-17idtfo-0 bIurki"
            label="Select"
            readonly=""
            tabindex="-1"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="StyledBox-sc-13pk1d4-0 eA-DBEo"
        style="min-width: auto;"
      >
        <svg
          aria-label="FormDown"
          class="StyledIcon-ofa7kd-0 fuviCN"
          viewBox="0 0 24 24"
        >
          <polyline
            fill="none"
            points="18 9 12 15 6 9"
            stroke="#000"
            stroke-width="2"
          />
        </svg>
      </div>
    </div>
  </button>
</div>
`;

exports[`Select selects an option with enter 1`] = `
<button
  aria-label="Open Drop"
  class="StyledButton-sc-323bzc-0 ebqXLG Select__StyledSelectDropButton-sc-17idtfo-1 iGEQF"
  id="test-select"
  type="button"
>
  <div
    class="StyledBox-sc-13pk1d4-0 jmuHez"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 iAryDB"
    >
      <div
        class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 dpaJQp"
      >
        <input
          autocomplete="off"
          class="StyledTextInput-sc-1x30a0s-0 fXMpul Select__SelectTextInput-sc-17idtfo-0 bIurki"
          id="test-select__input"
          placeholder="test select"
          readonly=""
          tabindex="-1"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="StyledBox-sc-13pk1d4-0 eA-DBEo"
      style="min-width: auto;"
    >
      <svg
        aria-label="FormDown"
        class="StyledIcon-ofa7kd-0 fuviCN"
        viewBox="0 0 24 24"
      >
        <polyline
          fill="none"
          points="18 9 12 15 6 9"
          stroke="#000"
          stroke-width="2"
        />
      </svg>
    </div>
  </div>
</button>
`;
